lib/validator.js

```diff
@@ -223,15 +271,28 @@ async function validateId() {
 	}
 
 	var credId = this.authnrData.get("credId");
-	if (credId !== undefined && !bufEqual(rawId, credId)) {
+	if (credId !== undefined && !abEqual(rawId, credId)) {
 		throw new Error("id and credId were not the same");
 	}
 
+	var allowCredentials = this.expectations.get("allowCredentials");
+
+	if (allowCredentials !== undefined) {
+		if (!allowCredentials.some(cred => {
+			var result = abEqual(rawId, cred.id);
+			return result;
+		}
+		)) {
+			throw new Error("id does not match any value in allowCredentials");
```

Comment:
Better: "Credential ID does not match [...]"

Reply:
Ok.